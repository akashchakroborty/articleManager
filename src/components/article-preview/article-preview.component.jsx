import React, { useState } from "react";
import { ArticlePreviewContainer } from "./article-preview.styles";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import SweetAlert from "react-bootstrap-sweetalert";
import ArticlePannel from "../article-pannel/article-pannel.component";
import ArticleImagePannel from "../article-image-pannel/article-image-pannel.component";
import {
  deleteArticle,
  getArticle,
} from "../../redux/articles/articles.action";

export const ArticlePreview = ({
  article,
  getArticle,
  deleteArticle,
  history,
  isArticleReadMore,
}) => {
  const { id, img, ...restProps } = article;
  const [showDeleteAlert, ToggleDeleteAlert] = useState(false);
  const handleReadClick = () => {
    history.push(`/readArticle/${id}`);
    getArticle(id);
  };
  const handleEditClick = () => {
    history.push(`editArticle/${id}`);
    getArticle(id);
  };
  const handleToggleDeleteClick = () => {
    console.log("true");
    ToggleDeleteAlert(true);
  };
  return (
    <ArticlePreviewContainer className="article-preview">
      <ArticlePannel
        handleReadClick={handleReadClick}
        handleEditClick={handleEditClick}
        handleToggleDeleteClick={handleToggleDeleteClick}
        isArticleReadMore={isArticleReadMore}
        {...restProps}
      />
      {showDeleteAlert && (
        <SweetAlert
          danger
          showCancel
          confirmBtnText="Yes, delete it!"
          confirmBtnBsStyle="danger"
          title="Are you sure you want to delete this article?"
          onConfirm={() => deleteArticle(id)}
          onCancel={() => {
            ToggleDeleteAlert(false);
          }}
          focusCancelBtn
        >
          You will not be able to recover this article again!
        </SweetAlert>
      )}
      <ArticleImagePannel
        img={img}
        handleReadClick={handleReadClick}
        isArticleReadMore={isArticleReadMore}
      />
    </ArticlePreviewContainer>
  );
};

export default withRouter(
  connect(null, { getArticle, deleteArticle })(React.memo(ArticlePreview))
);
