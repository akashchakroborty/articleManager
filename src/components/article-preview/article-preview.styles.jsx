import styled from "styled-components";

export const ArticlePreviewContainer = styled.div`
  width: 80%;
  display: flex;
  height: 340px;
  align-items: center;
  position: relative;
  margin-bottom: 40px;
  }
  @media screen and (max-width: 800px) {
    width: 100%;
    height: 300px
  }
`;

ArticlePreviewContainer.displayName = "ArticlePreviewContainer";
