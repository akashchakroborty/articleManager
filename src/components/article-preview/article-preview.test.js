import React from "react";
import { mount } from "enzyme";
import { ArticlePreview } from "./article-preview.component";

describe("ArticlePreview component", () => {
  let wrapper;
  let mockHistory;
  let mockGetArticle;
  let mockDeleteArticle;
  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };
    mockGetArticle = jest.fn();
    mockDeleteArticle = jest.fn(() => true);

    const mockProps = {
      article: {
        title: "Mock data",
        description: "Mock data",
        author: "Mock data",
        content: "Mock data",
        id: "Mock data",
        img: "Mock data",
        avatar: "Mock data",
        claps: "Mock data",
      },
      history: mockHistory,
      getArticle: mockGetArticle,
      deleteArticle: mockDeleteArticle,
    };

    wrapper = mount(<ArticlePreview {...mockProps} />);
  });

  it("should render ArticlePreview component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push when ArticleBody is clicked", () => {
    wrapper.find("ArticleBoby").simulate("click");
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it("should call history.push when Edit ActionContainer is clicked", () => {
    wrapper.find("ActionContainer").at(0).simulate("click");
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it("should call handleToggleDeleteClick when Delete ActionContainer is clicked", () => {
    const logSpy = jest.spyOn(console, "log");
    wrapper.find("ActionContainer").at(1).simulate("click");
    expect(logSpy).toHaveBeenCalledWith("true");
  });

  it(`should call handleToggleDeleteClick when Delete ActionContainer is clicked.
   If Yes Delete it is clicked, then deleteArticle is called.
   If cancel is clicked, ToggleDeleteClick is called and Sweet Alert is removed.
  `, () => {
    const logSpy = jest.spyOn(console, "log");
    wrapper.find("ActionContainer").at(1).simulate("click");
    expect(logSpy).toHaveBeenCalledWith("true");

    wrapper.find("SweetAlert .btn-danger").simulate("click");
    expect(mockDeleteArticle).toHaveBeenCalled();

    wrapper.find("SweetAlert .btn-link").simulate("click");
    expect(wrapper.exists("SweetAlert")).toBe(false);
  });
});
