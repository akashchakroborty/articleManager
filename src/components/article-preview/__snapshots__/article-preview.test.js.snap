// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreview component should render ArticlePreview component 1`] = `ReactWrapper {}`;
